Better management of the systemd services + workdir change

goldmann opened the pull request:
This commit includes a few fixes:

1. Adds `Requires` where it is justified
2. Adds the restart guards to all services
3. Changes the working directory for openshift-master service

Ad 2. For the SDN services `always` was used because of bug reported upstream: https://github.com/openshift/openshift-sdn/issues/12

Ad 3. The problem is that when vagrant boots the VM, it mounts the `/vagrant` directory when `openshift-master` is already running. This makes it impossible to remove the `openshift.local.*` directories to start with a fresh environment. This change moves the work dir to `/home/vagrant` to make it possible.

smarterclayton commented:
@rajatchopra review

rajatchopra commented:
@goldmann 
About restart guides, I think we should just rely on a fix to openshift/openshift-sdn#12 (coming soon!). 
Rest looks good.

goldmann commented:
@rajatchopra Right, but this doesn't hurt, correct? ;) I'm happy to change it to `on-failure` once the referenced issue gets fixed.

rajatchopra commented:
Ok, right. I do not mind this, since this localized to vagrant anyway. Will bug you to roll back to on-failure once we have a fix upstream.
Marking it [merge].

rajatchopra commented:
@goldmann 
Merge conflicts above.
Also, the fix for https://github.com/openshift/openshift-sdn/issues/12 is ready.
I think we do not need the restart logic anymore.
Update the PR please?

rajatchopra commented:
@goldmann 
I have removed the merge comment, meanwhile. We definitely do not need the restart logic anymore. 
https://github.com/openshift/openshift-sdn/pull/14 fixes that part.

goldmann commented:
Rebased,

goldmann commented:
Hold on with merging this PR.

goldmann commented:
I reworked once again this PR. The earlier proposal had this issues where there was no way to inject the certs into nodes. I switched back to use `/vagrant` dir for storing the certs, but I couldn't find any way to wait for the `/vagrant` directory to be mounted (tried `Requires=vagrant.mount` and `After=vagrant.mount` but it didn't worked as it should).

rajatchopra commented on vagrant/provision-master.sh:
```diff
@@ -37,13 +37,15 @@ systemctl start docker.service
 ExecStart=/usr/bin/openshift start master --public-master=${MASTER_IP} --nodes=${node_list}
 WorkingDirectory=/vagrant/
+Restart=on-failure
+RestartSec=10s
```
Do we need restart logic here? Any race conditions or just for catching unknowns?

goldmann replied:
The latter.

goldmann replied:
Do you want me to remove it?

rajatchopra replied:
Yes, my take would be to remove it. I think we want this to crash and burn if its not good :).

goldmann commented:
Removed the guard and rebased again.

rajatchopra commented:
[merge]
